library_project/books/serializers.py

```diff
@@ -0,0 +1,62 @@
+from rest_framework import serializers
+from .models import Book, BookReview
+import datetime
+import random
+
+class BookSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Book
+        fields = ('book_id', 'name', 'description', 'author', 'pages_count', 'genre')
+
+        def custom_validation(self):
+            # This function will send a validation error if the date of published is smaller then 0
+            
+            if self.date_of_published.year < 0:
+                raise serializers.ValidationError("The Date Of Published Has To Be Greater Then 0!")
+    
+    book_url = serializers.SerializerMethodField(source='get_book_url')
+
+
+    def get_book_url(self):
```

Comment:
one value in the input of this function is missing, the function return an error